src/kernel/filesystem/filesystem_common.zig

```diff
@@ -0,0 +1,173 @@
+const std = @import("std");
+
+// This is the assembly for the FAT bootleader.
+//     [bits   16]
+//     [org    0x7C00]
+//
+//     jmp     short _start
+//     nop
+//
+// times 87 db 0xAA
+//
+// _start:
+//     jmp     long 0x0000:start_16bit
+//
+// start_16bit:
+//     cli
+//     mov     ax, cs
+//     mov     ds, ax
+//     mov     es, ax
+//     mov     ss, ax
+//     mov     sp, 0x7C00
+//     lea     si, [message]
+// .print_string_with_new_line:
+//     mov     ah, 0x0E
+//     xor     bx, bx
+// .print_string_loop:
+//     lodsb
+//     cmp     al, 0
+//     je      .print_string_done
+//     int     0x10
+//     jmp     short .print_string_loop
+// .print_string_done:
+//     mov     al, 0x0A
+//     int     0x10
+//     mov     al, 0x0D
+//     int     0x10
+//
+// .reboot:
+//     xor     ah, ah
+//     int     0x16
+//     int     0x19
+//
+// .loop_forever:
+//     hlt
+//     jmp .loop_forever
+// message db "This is not a bootable disk. Please insert a bootable floppy and press any key to try again", 0
+// times 510 - ($ - $$) db 0
+// dw 0xAA55
+
+/// Basic boot code that will just print to the scream to insert a bootable image. This is intended
+/// as a place holder to be over written with real boot code if needed.
+/// This assumes the 512 sector size and includes the 0xAA55 boot signature.
+pub const bootsector_boot_code = [512]u8{
```

Comment:
Wouldn't it be more suitable for this to be in the mbr file as that is what uses it?

Reply:
It is also used in the FAT32 header, and maybe others for the boot sector.

Reply:
Hmm, since this is so specialised (we don't yet have other filesystems that use it) and is x86-specific, perhaps it should be moved to somewhere in arch/x86{_64}.

Reply:
I can do as it is x86 code. I don't know how this boot code is done for ARM, might do some testing for creating a FAT32/MBR partition on a ARM system.

Reply:
Nice :)

Reply:
> 
> 
> I can do as it is x86 code. I don't know how this boot code is done for ARM, might do some testing for creating a FAT32/MBR partition on a ARM system.

I'll make this a contributor friendly issue for the ARM boot sector